src/core.py

```diff
@@ -46,7 +51,10 @@ def on_config(self, config):
         # Theme
         theme_override = {}
         if "theme" in config:
-            theme_override = config["theme"].copy()
+            theme_override = config["theme"]
+            if config["theme"].name != "material":
```

Comment:
While testing this, I noticed that when no `theme` config is provided in an `mkdocs.yml`, the `config["theme"].name` value will always be `mkdocs` (the default).  This logic will likely need to be changed to account for that (or removed).